scheduler/test/cook/test/scheduler/dru.clj

```diff
@@ -120,6 +122,45 @@
                       shuffle
                       (into {})))))))))
 
+;; This test makes sure that running and queued jobs are sorted with the right
+;; implcit order, in particular, it makes sure the codepath of create-task-ent
+;; created fake instances for waiting jobs sort in the right order with respect
+;; to actual running instances by task->feature-vector
+(deftest test-sorted-task-scored-task-pairs-with-running
+  (setup)
+  (let [datomic-uri "datomic:mem://test-sorted-task-scored-task-pairs-order"
+        conn (restore-fresh-database! datomic-uri)
+        job1 (create-dummy-job conn :user "ljin" :memory 10.0 :ncpus 10.0 :name "1")
+        job2 (create-dummy-job conn :user "ljin" :memory 20.0 :ncpus 20.0 :name "2" :job-state :job.state/running)
+        job3 (create-dummy-job conn :user "ljin" :memory 40.0 :ncpus 40.0 :name "3")
+        job4 (create-dummy-job conn :user "ljin" :memory 80.0 :ncpus 80.0 :name "4" :job-state :job.state/running)
+        job5 (create-dummy-job conn :user "ljin" :memory 160.0 :ncpus 160.0 :name "5")
+        _ (create-dummy-instance conn job4 :instance-status :instance.status/running)
+        _ (create-dummy-instance conn job2 :instance-status :instance.status/running)
+        db (d/db conn)
+
+        pending-task-ents (->> (queries/get-pending-job-ents db)
+                               (map util/create-task-ent))
+        running-task-ents (util/get-running-task-ents db)
+        tasks (into (vec running-task-ents) pending-task-ents)]
+
+    (let [share {:mem 10.0 :cpus 10.0}
+          ; Queue should be job3, job5, becasue they're running, then 1 2 4.
+          ; DRU:     8.0 10.0 (not seen) 11.0 15.0 31.0
+          ordered-drus [8.0 10.0 11.0 15.0 31.0]]
+      (testing "dru order correct"
+        (is (= ordered-drus
+               (map (comp :dru second)
+                    (dru/sorted-task-scored-task-pairs
```

Comment:
This test checks that all 5 jobs we visit are in the cache.

Reply:
This test didn't exist until this PR.

If I ran this test without my cache change, the DRU order part would (and does) pass, because the DRU order shouldn't change based on cache.

The line below where I make sure the cache has 5 entries however would fail because it would only find 2 running jobs in the cache, not all 5. Now it passes.